lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -1210,6 +1288,408 @@ void BlockedEncodingAttr::print(mlir::AsmPrinter &printer) const {
   printer << "}>";
 }
 
+// FIXME Can we take the LinearLayout by const&?
+LogicalResult
+LinearEncodingAttr::verify(function_ref<InFlightDiagnostic()> emitError,
+                           LinearLayout linearLayout,
+                           ArrayRef<unsigned> repOrder) {
+  if (!isPermutationOfIota(repOrder)) {
+    return emitError() << "repOrder must be a permutation of 0..(rank-1). Got "
+                       << repOrder;
+  }
+  if (linearLayout.getNumOutDims() != repOrder.size()) {
+    return emitError() << "repOrder must have the same length as the number of "
+                       << "dimensions in the layout. Got " << repOrder
+                       << " for a layout with " << linearLayout.getNumOutDims()
+                       << " dimensions";
+  }
+
+  static const auto expectedInDims =
```

Comment:
Should we document that the input dimensions cannot be anything other than the following?

Reply:
Thinking about it a bit more. Maybe we also want to provide an example of linear layout's mlir representation above `let hasCustomAssemblyFormat=1`